homeassistant/components/elgato/light.py

```diff
@@ -112,22 +146,44 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         try:
             await self.elgato.light(on=False)
         except ElgatoError:
-            _LOGGER.error("An error occurred while updating the Elgato Key Light")
+            _LOGGER.exception("An error occurred while updating the Elgato Light")
```

Comment:
Will this print stack traces for connection errors too ? Could we create a decorator for the various services that will set it to unavailable if a connection error fails?

Reply:
> Will this print stack traces for connection errors too ?

Yes.

From the Python docs:

> Logs a message with level ERROR on this logger.
> Exception info is added to the logging message. This method should only be called from an exception handler.

(https://docs.python.org/3/library/logging.html#logging.Logger.exception)

Reply:
Example output:

```txt
2021-05-03 21:06:23 ERROR (MainThread) [homeassistant.components.elgato.light] An error occurred while updating the Elgato Light
Traceback (most recent call last):
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 999, in _create_direct_connection
    hosts = await asyncio.shield(host_resolved)
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 865, in _resolve_host
    addrs = await self._resolver.resolve(host, port, family=self._family)
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/aiohttp/resolver.py", line 31, in resolve
    infos = await self._loop.getaddrinfo(
  File "/home/frenck/.pyenv/versions/3.9.2/lib/python3.9/asyncio/base_events.py", line 856, in getaddrinfo
    return await self.run_in_executor(
  File "/home/frenck/.pyenv/versions/3.9.2/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/frenck/.pyenv/versions/3.9.2/lib/python3.9/asyncio/base_events.py", line 839, in _getaddrinfo_debug
    addrinfo = socket.getaddrinfo(host, port, family, type, proto, flags)
  File "/home/frenck/.pyenv/versions/3.9.2/lib/python3.9/socket.py", line 953, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/elgato/elgato.py", line 88, in _request
    response = await self._session.request(
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/aiohttp/client.py", line 520, in _request
    conn = await self._connector.connect(
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 535, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 892, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 1011, in _create_direct_connection
    raise ClientConnectorError(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host kill it:9123 ssl:default [Name or service not known]

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/frenck/code/home-assistant/core/homeassistant/components/elgato/light.py", line 179, in async_turn_on
    await self.elgato.light(
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/elgato/elgato.py", line 220, in light
    await self._request(
  File "/home/frenck/code/home-assistant/core/venv/lib/python3.9/site-packages/elgato/elgato.py", line 104, in _request
    raise ElgatoConnectionError(
elgato.exceptions.ElgatoConnectionError: Error occurred while communicating with Elgato Light device
```

Reply:
Sorry, I misformulated. I don't think that we should print a full stack trace for connection errors. Those are known errors and so should be handled accordingly.

Reply:
So I added this one because of feedback in the previous PR this week. (Don't get me wrong, I myself was perfectly happy with just the error message).

So let's say this happens, how would one get more information?

Reply:
Adjusted to what it was (which was fine IMHO)